calliope/core/preprocess/checks.py

```diff
@@ -212,8 +226,18 @@ def check_initial(config_model):
                 'be defined (tech: {})'.format(t_name)
             )
 
+    # Check whether any techs are erroneously in top-level location definitions
```

Comment:
I would definitely have preference for checking whether the loc_key subdict has any unexpected keys from `[techs, links, available_area, coordinates]` than this check, since it would cover someone not nesting a tech under `techs`, but misspelling that tech, for instance...

Reply:
Also, this same check should be added for links, right?

Reply:
Adding check for links, but a more complete solution will have to wait.